root/artist/ArtistRecordings.js

```diff
@@ -89,19 +93,35 @@ const ArtistRecordings = ({
           {show_all: `/artist/${artist.gid}/recordings?video=0`},
         )}
       </p>
-
     ) : (
-      <p>
-        {exp.l(
-          `Showing all recordings.
-           {show_sa|Show only standalone recordings instead}, or
-           {show_vid|show only videos}.`,
-          {
-            show_sa: `/artist/${artist.gid}/recordings?standalone=1`,
-            show_vid: `/artist/${artist.gid}/recordings?video=1`,
-          },
-        )}
-      </p>
+      hasStandalone && hasVideos ? (
+        <p>
+          {exp.l(
+            `Showing all recordings.
+            {show_sa|Show only standalone recordings instead}, or
+            {show_vid|show only videos}.`,
+            {
+              show_sa: `/artist/${artist.gid}/recordings?standalone=1`,
+              show_vid: `/artist/${artist.gid}/recordings?video=1`,
+            },
+          )}
+        </p>
+      ) : hasStandalone ? (
+        <p>
+          {exp.l(
+            `Showing all recordings.
+            {show_sa|Show only standalone recordings instead}.`,
+            {show_sa: `/artist/${artist.gid}/recordings?standalone=1`},
+          )}
+        </p>
+      ) : hasVideos ? (
+        <p>
+          {exp.l(
+            `Showing all recordings. {show_vid|Show only videos}.`,
+            {show_vid: `/artist/${artist.gid}/recordings?video=1`},
+          )}
+        </p>
+      ) : null
```

Comment:
It makes 3 localizable quite redundant localizable messages. If a 3rd parameter gets added, it would make 7. I would suggest to split into 3 separate sentences instead.

Reply:
Done. Since I was already making it modular, I expanded it to actually add the modular links to the standalone only and video only pages.